core/src/main/scala/com/cloudera/spark/CGroupsMemoryHandle.scala

```diff
@@ -0,0 +1,68 @@
+package com.cloudera.spark
+
+import com.cloudera.spark.CGroupsMemoryHandle.{countMetrics, metrics, getMetricsSnapshot}
+
+import scala.io.Source
+
+class CGroupsMemoryHandle extends MemoryGetter {
+
+  override val namesAndReporting = metrics.map { m =>
+    val peakReporting = if (countMetrics(m)) IncrementCounts else IncrementBytes
+    (m, peakReporting)
+  }
+
+  override def values(dest: Array[Long], offset: Int): Unit = {
+    var idx = offset
+    val metricsSnapshot = getMetricsSnapshot
+    metrics.foreach { m =>
+      metricsSnapshot.get(m).foreach { newValue =>
+        dest(idx) = newValue
+      }
+      idx += 1
+    }
+  }
+}
+
+object CGroupsMemoryHandle {
+
+  val cGroupMemoryStatPath = "/sys/fs/cgroup/memory/memory.stat"
+  // For more information about the metrics please visit
+  // https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/resource_management_guide/sec-memory
+  // https://docs.docker.com/config/containers/runmetrics/
+  val metrics = Seq("cache", "rss", "mapped_file", "pgpgin", "pgpgout", "swap", "active_anon", "inactive_anon",
+      "active_file", "inactive_file", "unevictable", "hierarchical_memory_limit", "hierarchical_memsw_limit")
+      .flatMap { metric => Seq(metric, "total_" + metric)}
```

Comment:
super nit (which doesn't really matter but again just getting you used to spark style): you double indent the continuation of the `Seq`, but only single indent the other code, the line w/ the `flatMap` in this case.  The point is to make the continuation more distinct from regular indentation.

I think this comes up more often when there is a long condition for an if --

```scala

if (some &&
    really &&
    long &&
    condition &&
    here) {
  regularCodeHere()
}
```

it makes the condition vs. the body more distinct

Reply:
Fixed in the latest commit